lib/jupyter/actions.js

```diff
@@ -381,6 +381,44 @@ define([
         );
       }
     }, 'extend-selection-below', 'vim-binding');
+
+    var cell_marker = {};
+
+    actions.register({
+      'handler': function(env, event) {
+        env.notebook.keyboard_manager.enabled = false
+        var oldCb = document.onkeydown;
+        var clear = () => {
+          document.onkeydown = oldCb;
+          env.notebook.keyboard_manager.enabled = true
+        };
+        document.onkeydown = function (event) {
+          clear();
+          var key = event.key;
+          var idx = env.notebook.get_selected_index();
+          cell_marker[key] = idx;
+        };
```

Comment:
I think using [addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) with `once: true` makes code much simpler.

Reply:
I know this implementation looks hacky, those who are already familiar with this Jupyter notebook extension are welcome to improve this. At last, this feature does not look too difficult to implement given a rich set of api we can leverage on inside `env.notebook` about managing the cell indices.

Reply:
I think I might have thought about that route you mentioned, I faced problems because `env.notebook.keyboard_manager` somehow over reaches the key signal, I worked around this by disabling it so that it only accepts the very next key signal coming from the user after pressing `m` or `~`

Reply:
Updated with { once: true } param and it is tested manually is working.